src/Adapters/EventSource.Net46/EventSourceListeningRequest.cs

```diff
@@ -0,0 +1,72 @@
+﻿// -----------------------------------------------------------------------
+// <copyright file="EventSourceListeningRequest.cs" company="Microsoft">
+// Copyright (c) Microsoft Corporation. 
+// All rights reserved.  2017
+// </copyright>
+// -----------------------------------------------------------------------
+
+using System.Diagnostics.Tracing;
+
+namespace Microsoft.ApplicationInsights.EventSourceListener
+{
+    /// <summary>
+    /// Represents a request to listen to specific EventSource
+    /// </summary>
+    public class EventSourceListeningRequest
+    {
+        /// <summary>
+        /// Gets or sets the name of the EventSource to listen to.
+        /// </summary>
+        public string Name { get; set; }
+
+        /// <summary>
+        /// Gets or sets the minimum level of an event that will be traced. 
+        /// </summary>
+        /// <remarks>
+        /// Events with level lower than the specified level will be silently discarded.
+        /// </remarks>
+        public EventLevel Level { get; set; }
+
+        /// <summary>
+        /// Gets or sets the keywords that must be set on an event to be included in tracing.
+        /// </summary>
+        public EventKeywords Keywords { get; set; }
+
+        /// <summary>
+        /// Initializes new instance of the <see cref="EventSourceListeningRequest"/> class
+        /// </summary>
+        /// <remarks>
+        /// By default all events from an EventSource are traced. The set of events can be restricted using <see cref="Level"/> and <see cref="Keywords"/> properties.
+        /// </remarks>
+        public EventSourceListeningRequest()
+        {
+            this.Level = EventLevel.LogAlways;
```

Comment:
Consistency is good and I agree that defaulting to Warning is good when it comes to tracing in production. 

But I will push back a bit here because with this default I am trying to optimize for easiest familiarization experience. The default Level value for EventAttribute is EventLevel.Informational. If we cut off everything below Warning by default, then events from simplest possible EventSource will not be captured, which might be tricky to figure out for a newbie. It is very easy to change the cutoff level through appinsights.config, so accepting everything seems like a good trade off.